src/storage-client/src/client.rs

```diff
@@ -354,6 +354,65 @@ impl PackableStats for SinkStatisticsUpdate {
     }
 }
 
+#[derive(Copy, Clone, Debug, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
+pub enum Status {
+    Starting,
+    Running,
+    Paused,
+    Stalled,
+    Ceased,
+    Dropped,
+}
+
+impl Status {
+    pub fn to_str(&self) -> &'static str {
+        match self {
+            Status::Starting => "starting",
+            Status::Running => "running",
+            Status::Paused => "paused",
+            Status::Stalled => "stalled",
+            Status::Ceased => "ceased",
+            Status::Dropped => "dropped",
+        }
+    }
+
+    /// Determines if a new status should be produced in context of a previous
+    /// status.
+    pub fn superseded_by(self, new: Status) -> bool {
```

Comment:
```suggestion
    pub fn can_supersede(self, new: Status) -> bool {
```

Reply:
`can_supersede` would be the inversion of this relationship––do you suggest I change something else about this?

Reply:
oh sorry!